src/workarounds/aarch64_mpi.jl

```diff
@@ -0,0 +1,90 @@
+"""
+MPI reduction opertations with custom types (i.e. anything that has not a MPI datatype equivalent)
+are not available on aarch64. These are temprorary workarounds, where variables with custom types 
+are broken down to standard types before communication, and recast to the initial types after.
+This file was created by fixing all MPI errors encountered by running the tests on an ARM machine:
+all sensible MPI reduction routines are implemented for each custom type causing an error.
+"""
+
+# Julia's Bool type has no direct equivalent MPI datatype => need integer conversion
+function mpi_min(bool::Bool, comm::MPI.Comm)
+    int = Int(bool)
+    Bool(mpi_min(int, comm))
+end
+
+function mpi_max(bool::Bool, comm::MPI.Comm)
+    int = Int(bool)
+    Bool(mpi_max(int, comm))
+end
+
+# Vec3{T} must be cast to Vector{T} before MPI reduction
+function mpi_sum!(arr::Vector{Vec3{T}}, comm::MPI.Comm) where{T}
+    n = length(arr)
+    new_arr = Vector{T}([])
+    for i in 1:n
+        append!(new_arr, arr[i][1:3])
+    end
+    mpi_sum!(new_arr, comm)
+    for i in 1:n
+	    arr[i] = Vec3{T}(new_arr[3(i-1)+1:3(i-1)+3])
+    end
+    arr
+end
+
+# ForwardDiff.Dual{T, U, V} and arrays of it must be cast to Vector{U} as well
+# utility function to cast a Dual type to an array containing a value and the partial diffs
+function dual_array(dual::ForwardDiff.Dual{T, U, V}) where{T, U, V}
+    dual_array = [ForwardDiff.value(dual)]
+    append!(dual_array, collect(ForwardDiff.partials(dual)))
```

Comment:
I vaguely recall there are again ways to do thiy without copy. Check the forwarddiff rules for array operations (multiplication neefs to be done equally to all partials)